File: examples/openai_client.js
#!/usr/bin/env node

/**
 * Example OpenAI client for Rust LLM Runner (Node.js)
 */

import OpenAI from 'openai';

const client = new OpenAI({
  baseURL: 'http://localhost:11434/v1',
  apiKey: 'not-needed',
});

async function chatExample() {
  console.log('=== Chat Completion Example ===\n');
  
  const response = await client.chat.completions.create({
    model: 'llama4:scout',
    messages: [
      { role: 'system', content: 'You are a helpful assistant.' },
      { role: 'user', content: 'What is Rust programming language?' }
    ],
    temperature: 0.7,
    max_tokens: 500,
  });
  
  console.log(response.choices[0].message.content);
  console.log(`\nTokens used: ${response.usage.total_tokens}`);
}

async function streamingExample() {
  console.log('\n=== Streaming Example ===\n');
  
  const stream = await client.chat.completions.create({
    model: 'llama4:scout',
    messages: [
      { role: 'user', content: 'Write a short poem about coding' }
    ],
    stream: true,
  });
  
  for await (const chunk of stream) {
    const content = chunk.choices[0]?.delta?.content;
    if (content) {
      process.stdout.write(content);
    }
  }
  
  console.log('\n');
}

async function main() {
  try {
    await chatExample();
    await streamingExample();
  } catch (error) {
    console.error('Error:', error.message);
  }
}

main();

File: examples/openai_client.py
#!/usr/bin/env python3
"""
Example OpenAI client for Rust LLM Runner
"""

from openai import OpenAI

client = OpenAI(
    base_url="http://localhost:11434/v1",
    api_key="not-needed"
)

def chat_example():
    print("=== Chat Completion Example ===\n")
    
    response = client.chat.completions.create(
        model="llama4:scout",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": "What is Rust programming language?"}
        ],
        temperature=0.7,
        max_tokens=500
    )
    
    print(response.choices[0].message.content)
    print(f"\nTokens used: {response.usage.total_tokens}")

def streaming_example():
    print("\n=== Streaming Example ===\n")
    
    stream = client.chat.completions.create(
        model="llama4:scout",
        messages=[
            {"role": "user", "content": "Write a short poem about coding"}
        ],
        stream=True
    )
    
    for chunk in stream:
        if chunk.choices[0].delta.content:
            print(chunk.choices[0].delta.content, end='', flush=True)
    
    print("\n")

if __name__ == "__main__":
    chat_example()
    streaming_example()
